webapp/app/forms/steps/eligibility_steps.py

```diff
@@ -29,106 +114,395 @@ def _main_handle(self, stored_data):
-                                  'text': _l('form.eligibility.income-kapital-mit-pauschalbetrag-help')}})
-        kapitaleink_ohne_steuerabzug = YesNoField(_l('form.eligibility.income-kapital-ohne-steuerabzug'),
-            render_kw={'detail': {'title': _l('form.eligibility.income-kapital-ohne-steuerabzug.detail.title'),
-                                  'text': _l('form.eligibility.income-kapital-ohne-steuerabzug-help')}})
-        kapitaleink_guenstiger = YesNoField(_l('form.eligibility.income-kapital-neu'),
-            render_kw={'detail': {'title': _l('form.eligibility.income-kapital-neu.detail.title'),
-                                  'text': _l('form.eligibility.income-kapital-neu-help')}})
-        geringf = YesNoField(_l('form.eligibility.income-geringf'))
-        erwerbstaetigkeit = YesNoField(_l('form.eligibility.income-erwerbstaetigkeit'))
-        unterhalt = YesNoField(_l('form.eligibility.income-unterhalt'))
-        ausland = YesNoField(_l('form.eligibility.income-ausland'),
-            render_kw={'detail': {'title': _l('form.eligibility.income-ausland.detail.title'),
-                                  'text': _l('form.eligibility.income-ausland-help')}})
-        other = YesNoField(_l('form.eligibility.income-other'))
-        verheiratet_zusammenveranlagung = YesNoField(_l('form.eligibility.verheiratet-zusammenveranlagung'))
-        verheiratet_einzelveranlagung = YesNoField(_l('form.eligibility.verheiratet-einzelveranlagung'))
-        geschieden_zusammenveranlagung = YesNoField(_l('form.eligibility.geschieden-einzelveranlagung'))
-        elster_account = YesNoField(_l('form.eligibility.elster-account'),
-            render_kw={'detail': {'title': _l('form.eligibility.elster-account.detail.title'),
-                                  'text': _l('form.eligibility.elster-account-help')}})
-
-    def __init__(self, endpoint, **kwargs):
-        super(EligibilityIncomesFormSteuerlotseStep, self).__init__(
-            form=self.IncomesForm,
-            endpoint=endpoint,
-            header_title=_('form.eligibility.header-title'),
-            **kwargs,
-            )
+    class InputForm(SteuerlotseBaseForm):
+        marital_status_eligibility = RadioField(
+            label=_l('form.eligibility.marital_status'),
+            render_kw={'hide_label': True},
+            choices=[('married', _l('form.eligibility.marital_status.married')),
+                     ('widowed', _l('form.eligibility.marital_status.widowed')),
+                     ('single', _l('form.eligibility.marital_status.single')),
+                     ('divorced', _l('form.eligibility.marital_status.divorced')),
+                     ])
 
     def _main_handle(self, stored_data):
         stored_data = super()._main_handle(stored_data)
-        self.render_info.additional_info['next_button_label'] = _('form.eligibility.send-button')
+        stored_data = dict(filter(lambda elem: elem[0] == 'marital_status_eligibility', stored_data.items()))
```

Comment:
Only the MaritalStatusInputStep does not have a recursive data_model. Therefore, it is not possible to use the same function as the one you are mentioning (which isn't part of the parent step of this one anyway). However, we could restructure this a little bit and then have it here in the same way. I did not do this, because this is the first step and works a little different as it does not rely on the others. 

While thinking about it, we could put this into the parent class. and then have the decision input step override the function?

Reply:
We've decided to add an additional step for multiple decisions and use that one for the marital status input. This step will share a lot more with the decision steps then.